pkg/util/installconfig/modules.go

```diff
@@ -44,27 +50,46 @@ type Modules struct {
 }
 
 func GetModules() Modules {
+	if override != nil {
+		return *override
+	}
+
 	once.Do(func() {
-		modulesJson := os.Getenv(modulesJsonEnv)
+		modulesJson := os.Getenv(ModulesJsonEnv)
 		if modulesJson == "" {
-			log.Info("envvar not set, using default", "envvar", modulesJsonEnv)
+			log.Info("envvar not set, using default", "envvar", ModulesJsonEnv)
 
 			modules = fallbackModules
 		}
 
 		err := json.Unmarshal([]byte(modulesJson), &modules)
 		if err != nil {
-			log.Info("problem unmarshalling envvar content, using default", "envvar", modulesJsonEnv, "err", err)
+			log.Info("problem unmarshalling envvar content, using default", "envvar", ModulesJsonEnv, "err", err)
 
 			modules = fallbackModules
 		}
 
-		log.Info("envvar content read and set", "envvar", modulesJsonEnv, "value", modulesJson)
+		log.Info("envvar content read and set", "envvar", ModulesJsonEnv, "value", modulesJson)
 	})
 
 	return modules
 }
 
+// SetModulesOverride is a testing function, so you can easily unittest function using the GetModules() func
+func SetModulesOverride(t *testing.T, modules Modules) {
+	t.Helper()
+
+	override = &modules
+
+	t.Cleanup(func() {
+		override = nil
+	})
+}
+
 func GetModuleValidationErrorMessage(moduleName string) string {
 	return fmt.Sprintf(validationErrorTemplate, moduleName, moduleName)
 }
+
+func GetCSIValidationErrorMessage() string {
+	return fmt.Sprintf(validationErrorTemplate, "csi-driver", "CloudNative")
```

Comment:
> csi-driver has been disabled during Operator install. The necessary resources for CloudNative to work are not present on the cluster. Redeploy the Operator via Helm with all the necessary resources enabled.

CSI Driver was disabled during Operator install. The necessary resources for Cloud Native Fullstack to work are not present on the cluster. Redeploy the Operator via Helm with all the necessary resources enabled.

Reply:
I appreciate the attempt to make this as generic as possible, but it's kinda hard to judge what I'm supposed to do as a user. Can we somehow fit the relevant `values.yaml` section in here or is that too detailed?

Reply:
on it

Reply:
Adding the helm value in the message can get out of sync